mqtt/mqttd/src/main.rs

```diff
@@ -23,6 +23,16 @@ async fn main() -> Result<(), Error> {
         .nth(1)
         .unwrap_or_else(|| "0.0.0.0:1883".to_string());
 
+    // TODO pass it to broker
+    // TODO make it an argument to override defaul config
+    let path: Option<String> = None;
+    let config = path
```

Comment:
Interesting, for iotedged we used a different approach, where we have a method like `new` that takes `Option<Path>` rather than 2 almost identical methods. Any benefits to have 2 methods?